AirCasting/Utils/CLLocationUtils.swift

```diff
@@ -0,0 +1,35 @@
+// Created by Lunar on 14/06/2021.
+//
+
+import CoreLocation
+
+extension CLLocationCoordinate2D: Equatable { }
+
+extension CLLocationCoordinate2D {
+    public init?(latitude: Double?, longitude: Double?) {
+        guard let lat = latitude, let long = longitude else { return nil }
+        self.init(latitude: .init(lat), longitude: .init(long))
+    }
+}
+
+public func ==(lhs: CLLocationCoordinate2D, rhs: CLLocationCoordinate2D) -> Bool {
+    return lhs.latitude == rhs.latitude && lhs.longitude == rhs.longitude
+}
+
+extension Optional where Wrapped == Double {
+    var locationDegrees: CLLocationDegrees {
+        guard case .some(let value) = self else { return .zero }
+        return CLLocationDegrees(value)
+    }
+}
+
+extension CLLocationCoordinate2D {
+    //
+    // https://en.wikipedia.org/wiki/Null_Island
+    //
+    // since 0 is exactly representable as an IEEE754 floating-point number
+    // comparison with 0 should be safe here:
+    var isLocationLess: Bool {
+        return latitude == 0.0 && longitude == 0.0
```

Comment:
This is not location less. 0 0 is a place in the word. Its in the middle of the sea east to Africa. 
Also please consider checking for CLLocationInvalid. coordinate system only accepts values : lat -90...90 and long -180...180. But in some cases the Aircasting logic may return a location 200,200 which is an invalid value and should be used only as fallback and mapped to nil location in our system.

Reply:
I was not sure what locationLess means in API calls, I think I got it wrong, will remove 👌